pkg/controllers/canary_controller.go

```diff
@@ -225,3 +243,27 @@ func (r *CanaryReconciler) includeNamespace(namespace string) bool {
 	}
 	return false
 }
+
+func getCheckIDsForCanary(canaryID uuid.UUID) []string {
+	checks, _ := db.GetAllChecksForCanary(canaryID)
+	var checkIDs []string
+	for _, check := range checks {
+		checkIDs = append(checkIDs, check.ID)
+	}
+	return checkIDs
+}
+
+// getRemovedChecks returns the checkIds which are present in status but not present in the canary.
+func getRemovedChecks(a, b []string) []string {
+	mb := make(map[string]struct{}, len(b))
+	for _, x := range b {
+		mb[x] = struct{}{}
+	}
+	var diff []string
+	for _, x := range a {
+		if _, found := mb[x]; !found {
+			diff = append(diff, x)
+		}
+	}
+	return diff
```

Comment:
We can rename it to `SetDifference` and treat it like a utility function

Reply:
No need for giving it the checks and status context

Reply:
yeah, make sense. moved io to utils